sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs

```diff
@@ -362,14 +377,15 @@ public void CreatesManagedServiceIdentityCredentialsWithClientId()
             Assert.IsInstanceOf<ManagedIdentityCredential>(credential);
             var managedIdentityCredential = (ManagedIdentityCredential)credential;
 
-            var client = (ManagedIdentityClient)typeof(ManagedIdentityCredential).GetProperty("Client", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(managedIdentityCredential);
-            var clientId = typeof(ManagedIdentityClient).GetProperty("ClientId", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(client);
+            string clientId;
+            int idType;
+            ReflectIdAndType(managedIdentityCredential, out clientId, out idType);
```

Comment:
Curiosity only: why do we need reflection here?  Can we not make a runtime type check?

Reply:
Because these properties and fields are internal or private.